wallets/react-wallet-chat/src/pages/newChat.tsx

```diff
@@ -1,47 +1,72 @@
-import { Fragment, useEffect, useState } from 'react'
+import { Fragment, useCallback, useEffect, useState } from 'react'
 import { useSnapshot } from 'valtio'
 import { FiArrowRight } from 'react-icons/fi'
 import { Input, Row } from '@nextui-org/react'
-
 import PageHeader from '@/components/PageHeader'
 import { chatClient } from '@/utils/WalletConnectUtil'
 import { ChatClientTypes } from '@walletconnect/chat-client'
 import ChatPrimaryCTAButton from '@/components/ChatPrimaryCTAButton'
 import { demoContactsMap } from '@/config/chatConstants'
 import SettingsStore from '@/store/SettingsStore'
+import { useRouter } from 'next/router'
 
 export default function NewChatPage() {
   const [address, setAddress] = useState('')
+  const router = useRouter()
+
   const { eip155Address } = useSnapshot(SettingsStore.state)
 
-  const createInvite = async (targetAddress: string) => {
-    const invite: ChatClientTypes.PartialInvite = {
-      message: "hey let's chat",
-      account: `eip155:1:${eip155Address}`
-    }
-    const inviteId = await chatClient.invite({
-      account: targetAddress,
-      invite
+  useEffect(() => {
+    chatClient.on('chat_joined', args => {
+      const newChatTarget = new URLSearchParams(document.location.search).get('target')
+      router.push(`/chat?topic=${args.topic}&peerAccount=${newChatTarget}`)
     })
-  }
+  }, [router])
+
+  const createInvite = useCallback(
+    async (targetAddress: string) => {
+      const invite: ChatClientTypes.PartialInvite = {
+        message: "hey let's chat",
+        account: `eip155:1:${eip155Address}`
+      }
+      await chatClient.invite({
+        account: targetAddress,
+        invite
+      })
+    },
+    [eip155Address]
+  )
+
+  const onInvite = useCallback(
+    async (addressToInvite: string) => {
+      if (demoContactsMap[addressToInvite]) {
```

Comment:
Not in scope for this PR, but we can kill this whole demo mapping at some point. This was just for the MVP/demo video.

Reply:
For sure, something to tackle soon